crates/task-impls/src/quorum_vote/mod.rs

```diff
@@ -133,7 +134,7 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES> + 'static> VoteDependencyHand
                 &parent,
                 &proposed_leaf.block_header().clone(),
                 vid_share.data.common.clone(),
-                self.version,
+                *self.version.read().await,
```

Comment:
I think there's a danger we get versions mixed up here, if we use a `version` RwLock rather than calculating it dynamically from the `decided_upgrade_certificate` and the view number -- this would probably only happen around an upgrade view, if there's enough network latency where we receive some messages out of order.

I think this is fine for now, just noting that we should definitely switch to using `decided_upgrade_certificate`  + view number to decide the version.